sqlx-macros/src/query/mod.rs

```diff
@@ -309,14 +307,13 @@ where
     // If the build is offline, the cache is our input so it's pointless to also write data for it.
     #[cfg(feature = "offline")]
     if !offline {
-        let mut save_dir = std::path::PathBuf::from(
-            env::var("CARGO_TARGET_DIR").unwrap_or_else(|_| "target/".into()),
-        );
-
-        save_dir.push("sqlx");
+        let save_dir = env::var_os("CARGO_MANIFEST_DIR")
+            .map(PathBuf::from)
+            .unwrap_or_else(|| Path::new(".").to_owned())
+            .join(".sqlx");
 
         std::fs::create_dir_all(&save_dir)?;
-        data.save_in(save_dir, input.src_span)?;
+        data.save_in(save_dir)?;
```

Comment:
It's poor behavior to automatically create folders outside of `target/`. IMO, we shouldn't do anything if the folder isn't there.

Reply:
I understand the desire to not create folders/files outside of `target/` but in my mind the implicit nature is important for the developer experience. 

As a compromise, we could use `target/sqlx` but have a simple way of changing that folder as an "opt-in" to committing your query data.

Reply:
> As a compromise, we could use `target/sqlx` but have a simple way of changing that folder as an "opt-in" to committing your query data.

What way are you imagining? Thinking about this, if one wanted to only have the cache for a subset of crates in a workspace with a bunch of crates using SQLx, it wouldn't be so simple. Maybe `cargo sqlx prepare` could warn about crates that depend on SQLx but don't have a `.sqlx` dir next to their `Cargo.toml` (and error if this is the case for all of them)?

Reply:
I think more simply we just don't do anything if `.sqlx` doesn't exist, and `cargo sqlx prepare` creates it if it doesn't and then runs compilation.